x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -93,12 +95,14 @@ public DeploymentManager(
         NamedXContentRegistry xContentRegistry,
         ThreadPool threadPool,
         PyTorchProcessFactory pyTorchProcessFactory,
-        int maxProcesses
+        int maxProcesses,
+        SystemAuditor systemAuditor
```

Comment:
It might be more appropriate to use the `InferenceAuditor` rather than the `SystemAuditor`. Creating and deleting a trained model uses the `InferenceAuditor` as does the model download notifications. SystemAuditor is used when rebalancing the deployments after a cluster change. Usually when the message applies to a single model/deployment the InferenceAuditor is used